blocks/graph.py

```diff
@@ -520,3 +521,82 @@ def apply_dropout(computation_graph, variables, drop_prob, rng=None,
         replacement.tag.replacement_of = variable
 
     return computation_graph.replace(replacements)
+
+
+def apply_batch_normalization(computation_graph, variables, gammas,
+                              betas, axis=0, epsilon=1e-7):
+    """Returns a graph to variables in a computational graph.
+
+    Parameters
+    ----------
+    computation_graph : instance of :class:`ComputationGraph`
+        The computation graph.
+    variables : list of :class:`~tensor.TensorVariable`
```

Comment:
Reading the paper, I don't know.

For fully-connected networks what I understand is that they batch-normalize post-nonlinearity activations (this is what I read as "activations", but I might be wrong).

For convnets the authors explicitly say that they batch-normalize the post-convolution, pre-nonlinearity activations.

I could put general recommendations in the documentation (after validating my understanding) but I wouldn't go as far as making assumptions for the user.